aptos-move/framework/aptos-framework/sources/AuctionHouse.move

```diff
@@ -0,0 +1,209 @@
+                auction_items: Table::new<TokenId, AuctionItem>(),
+                auction_events: Event::new_event_handle<AuctionEvent>(sender),
+                bid_events: Event::new_event_handle<BidEvent>(sender),
+                claim_coins_events: Event::new_event_handle<ClaimCoinsEvent>(sender),
+                claim_token_events: Event::new_event_handle<ClaimTokenEvent>(sender),
+            });
+        };
+        let start_time = Timestamp::now_microseconds();
+        let token = Token::withdraw_token(sender, token_id, 1);
+
+        let auction_data = borrow_global_mut<AuctionData>(sender_addr);
+        let auction_items = &mut auction_data.auction_items;
+
+        // if auction_items still contain token_id, this means that when sender_addr last auctioned this token,
+        // they did not claim the coins from the highest bidder
+        // sender_addr has received the same token somehow but has not claimed the coins from the initial auction
+        assert!(!Table::contains(auction_items, token_id), ERROR_CLAIM_COINS_FIRST);
+
+        Event::emit_event<AuctionEvent>(
+            &mut auction_data.auction_events,
+            AuctionEvent { id: token_id, min_selling_price: min_selling_price, duration: duration },
+        );
+
+        Table::add(auction_items, token_id, AuctionItem {
+            min_selling_price,
+            duration,
+            start_time,
+            current_bid: min_selling_price - 1,
+            current_bidder: sender_addr,
+            locked_token: Option::some(token),
+        })
+    }
+
+    fun is_auction_active(start_time: u64, duration: u64): bool {
+        let current_time = Timestamp::now_microseconds();
+        current_time <= start_time + duration && current_time >= start_time
+    }
+
+    fun is_auction_complete(start_time: u64, duration: u64): bool {
+        let current_time = Timestamp::now_microseconds();
+        current_time > start_time + duration
+    }
+
+    public(script) fun bid(sender: &signer, seller: address, creator: address, collection_name: vector<u8>, name: vector<u8>, bid: u64) acquires CoinEscrow, AuctionData {
+        let token_id = Token::create_token_id_raw(creator, collection_name, name);
+        let sender_addr = Signer::address_of(sender);
+        assert!(sender_addr != seller, ERROR_INVALID_BUYER);
+
+        let auction_data = borrow_global_mut<AuctionData>(seller);
+        let auction_items = &mut auction_data.auction_items;
+        let auction_item = Table::borrow_mut(auction_items, token_id);
+        assert!(is_auction_active(auction_item.start_time, auction_item.duration), ERROR_AUCTION_INACTIVE);
+
+        assert!(bid > auction_item.current_bid, ERROR_INSUFFICIENT_BID);
+
+        if (!exists<CoinEscrow>(sender_addr)) {
+            move_to(sender, CoinEscrow {
+                locked_coins: Table::new<TokenId, Coin<TestCoin>>()
+            });
+        };
+
+        if (auction_item.current_bidder != seller) {
```

Comment:
I don't understand this. I think this should apply for any user. I think what you want to check is there actually a current bid at start?